nifi-api/src/main/java/org/apache/nifi/components/PropertyDescriptor.java

```diff
@@ -396,6 +396,20 @@ public <E extends Enum<E>> Builder allowableValues(final E[] values) {
             return this;
         }
 
+        /**
+         * Stores allowable values from an enum class.
+         * @param enumClass an enum class that implements the Allowable interface and contains a set of values
+         * @param <E> generic parameter for an enum class that implements the Allowable interface
+         * @return the builder
+         */
+        public <E extends Enum<E> & DescribedValue> Builder allowableValues(final Class<E> enumClass) {
+            this.allowableValues = new ArrayList<>();
+            for (E enumValue : enumClass.getEnumConstants()) {
+                this.allowableValues.add(new AllowableValue(enumValue.name(), enumValue.getDisplayName(), enumValue.getDescription()));
```

Comment:
Following the interface adjustments, `enumValue.name()` should be replaced with `enumValue.getValue()`:

```suggestion
                this.allowableValues.add(new AllowableValue(enumValue.getValue(), enumValue.getDisplayName(), enumValue.getDescription()));
```

Reply:
Oh wow, thanks!